src/components/Forms/create-repair-request.tsx

```diff
@@ -0,0 +1,169 @@
+// Page for submitting a repair request
+// This form is copied from /repair-request/index.tsx
+import Image from "next/image";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { SubmitHandler, useForm } from "react-hook-form";
+import { z } from "zod";
+
+import Button from "@/components/Button";
+import FieldImageUpload from "@/components/FormFields/field-image-upload";
+import SingleSelect from "@/components/FormFields/field-single-select";
+import FieldTextArea from "@/components/FormFields/field-text-area";
+import { TermsAndConditions } from "@/components/terms-and-conditions";
+import { Brand, useBrands } from "@/hooks/brands";
+import { EventOption, useEventOptions } from "@/hooks/events";
+import { ItemType, useItemTypes } from "@/hooks/item-types";
+import { useCreateRepairRequest } from "@/hooks/repair-request";
+import generateThumbnail from "@/lib/gen-thumbnail";
+import uploadImage from "@/lib/upload-image";
+import { createRepairRequestSchema } from "@/schema/repair-request";
+import { CreateRepairRequest } from "@/types";
+
+export interface FormValues extends CreateRepairRequest {
+  tncAccepted: boolean;
+}
+
+const repairRequestFormSchema = createRepairRequestSchema.extend({
+  tncAccepted: z.boolean().refine((val) => !!val, {
+    message: "Please read through and accept the house rules."
+  })
+});
+
+const Home = () => {
+  const { control, handleSubmit, setValue } = useForm<FormValues>({
+    resolver: zodResolver(repairRequestFormSchema),
+    defaultValues: {
+      itemBrand: "",
+      itemType: "",
+      description: "",
+      images: [],
+      eventId: "",
+      comment: "",
+      tncAccepted: false
+    }
+  });
+
+  const { data: itemTypeList } = useItemTypes();
+  const { data: brandList } = useBrands();
+  const { data: eventOptions } = useEventOptions();
+  const { mutate: createRepairRequest } = useCreateRepairRequest();
+
+  const onSubmit: SubmitHandler<FormValues> = async (data) => {
+    const uploadPromises =
+      data.images?.map((image) => uploadImage(image)) ?? [];
+    const uploadThumbailPromise = data.images?.[0]
+      ? generateThumbnail(data.images[0]).then(uploadImage)
+      : undefined;
+    const keys = await Promise.all([uploadThumbailPromise, ...uploadPromises]);
+    const [thumbnailImage, ...images] = keys;
+    const updatedData = { ...data, thumbnailImage, images };
+    createRepairRequest(updatedData);
+  };
+
+  return (
+    <div className="flex items-center justify-center p-4">
+      <div className="flex w-screen flex-col justify-center gap-4 md:w-3/6 lg:w-full">
+        {/* Logo of Repair Lab, which links to the main website. */}
+
+        <picture className="flex justify-center">
```

Comment:
This component should have the form and the functionality for it to work -- and not have any other UI bits in it.

Reply:
Additionally ![image](https://github.com/codersforcauses/repair-labs/assets/19729435/a5f183b2-a940-42fb-9f37-a6752d526d3a)